### PR TITLE
Add builders for MySQL, PostgreSQL, BigQuery query runners

### PR DESCRIPTION
**Pull request by findepi**
Avoid multiple overloads. Make setup easier to follow.

extracted from https://github.com/trinodb/trino/pull/21588

### CODE REVIEW

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java [hunk 1]
```diff
@@ -62,67 +61,69 @@ public final class BigQueryQueryRunner
...
         }
-        catch (Throwable e) {
-            closeAllSuppress(e, queryRunner);
-            throw e;
+
+        @CanIgnoreReturnValue
+        public Builder setConnectorProperties(Map<String, String> connectorProperties)
+        {
+            this.connectorProperties = connectorProperties;
+            return this;
```
Comment (wendigo) [inline plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java:87]: `return self();`

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/MySqlQueryRunner.java [hunk 2]
```diff
@@ -23,70 +23,74 @@
...
-            // note: additional copy via ImmutableList so that if fails on nulls
-            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
-            connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
-            connectorProperties.putIfAbsent("connection-user", server.getUsername());
-            connectorProperties.putIfAbsent("connection-password", server.getPassword());
+        @CanIgnoreReturnValue
+        public Builder addConnectorProperties(Map<String, String> connectorProperties)
+        {
+            this.connectorProperties.putAll(connectorProperties);
+            return this;
```
Comment (wendigo) [inline plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/MySqlQueryRunner.java:65]: return self();

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/PostgreSqlQueryRunner.java [hunk 3]
```diff
@@ -23,72 +24,79 @@
...
         }
-        catch (Throwable e) {
-            closeAllSuppress(e, queryRunner, server);
-            throw e;
+
+        @CanIgnoreReturnValue
+        public Builder addConnectorProperties(Map<String, String> connectorProperties)
+        {
+            this.connectorProperties.putAll(connectorProperties);
+            return this;
```
Comment (wendigo) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/PostgreSqlQueryRunner.java:67]: return self();

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 4]
```diff
@@ -145,11 +144,9 @@ protected QueryRunner createQueryRunner()
             throws Exception
     {
         postgreSqlServer = closeAfterClass(new TestingPostgreSqlServer());
-        return createPostgreSqlQueryRunner(
-                postgreSqlServer,
-                ImmutableMap.of(),
-                ImmutableMap.of("jdbc-types-mapped-to-varchar", "Tsrange, Inet" /* make sure that types are compared case insensitively */),
-                ImmutableList.of());
+        return PostgreSqlQueryRunner.builder(postgreSqlServer)
+                .addConnectorProperties(Map.of("jdbc-types-mapped-to-varchar", "Tsrange, Inet" /* make sure that types are compared case insensitively */))
```
Comment (wendigo) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java:148]: nit: this comment would look better above the .addConnectorProperties call

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java:87]: `return this` in the other method

**Pull request merged**